tests/check/check_signed_video_auth.c

```diff
@@ -2188,6 +2188,84 @@ START_TEST(with_blocked_signing)
 }
 END_TEST
 
+/* Test description
+ * Generates SEIs using golden SEI prinsiple and verifies them.
+ * The operation is as follows:
+ * 1. Generate and store golden SEI
+ * 2. Enable golden SEI principle
+ * 3. Introduce a I frame to generate a new SEI frame
+ * 4. Get the SEI frame
+ * 5. Verify the stored golden SEI and newly fetched SEI
+ */
+START_TEST(golden_sei_principle)
+{
+
+  SignedVideoCodec codec = settings[_i].codec;
+  SignedVideoReturnCode sv_rc;
+  signed_video_t *sv = signed_video_create(codec);
+  ck_assert(sv);
+  char *private_key = NULL;
+  size_t private_key_size = 0;
+  // Setup the key
+  sv_rc = settings[_i].generate_key(NULL, &private_key, &private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  sv_rc = signed_video_set_private_key_new(sv, private_key, private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_set_hash_algo(sv, settings[_i].hash_algo_name);
```

Comment:
Just like you apply the hash algo, you should apply other appropriate settings from `settings[_i]` which I guess is only the authenticity level